pages/cql/bcs-logic-2018.cql

```diff
@@ -64,72 +91,120 @@ define "Is Age 52 to 74 at End":
-	// marked as a stub / draft / incomplete.
-
-define "Is In Eligible Population Medicaid":
-	"Is In Eligible Population"
-		and "Is Continuous Enrollment Medicaid"
-
-define "Is Continuous Enrollment Medicaid":
-	true
-	// TODO: Determine continuous enrollment over the 39 month period consisting
-	// of the measurement year plus the 2 prior years plus the 3 prior months.
-	// No gaps in enrollment are allowed the first Oct 1-Dec 31 of that 39 month period. 
-	// Allowable gap -- No more than one gap in enrollment of up to 45 days
-	// for each full calendar year of continuous enrollment (i.e., the
-	// measurement year and the year prior to the measurement year).
-	// To determine continuous enrollment for a Medicaid beneficiary for
-	// whom enrollment is verified monthly, the member may not have more
-	// than a 1-month gap in coverage during each full calendar year of
-	// continuous enrollment (i.e., the measurement year and the year prior
-	// to the measurement year).
-	// See http://hl7.org/fhir/enrollmentresponse.html which is currently
-	// marked as a stub / draft / incomplete.
-
-define "Is In Eligible Population Medicare":
-	"Is In Eligible Population"
-		and "Is Continuous Enrollment Medicare"
-		and (if "Is Age 65 Plus at Start"
-			then ("Is Enrolled in Institutional SNP"
-				or "Is Living Long-Term in Institution")
-			else false)
-
-define "Is Continuous Enrollment Medicare":
-	true
-	// TODO: Determine continuous enrollment over the 39 month period consisting
-	// of the measurement year plus the 2 prior years plus the 3 prior months.
-	// No gaps in enrollment are allowed the first Oct 1-Dec 31 of that 39 month period. 
-	// Allowable gap -- No more than one gap in enrollment of up to 45 days
-	// for each full calendar year of continuous enrollment (i.e., the
-	// measurement year and the year prior to the measurement year).
-	// See http://hl7.org/fhir/enrollmentresponse.html which is currently
-	// marked as a stub / draft / incomplete.
+define "Is In Applicable Product Line":
+	case
+		when ("Product Line" ~ null) then true
+		when ("Product Line" = 'commercial') then "Is Continuous Enrollment"
+		when ("Product Line" = 'medicaid') then "Is Continuous Enrollment"
+		when ("Product Line" = 'medicare') then
+			("Is Continuous Enrollment"
+				and (if "Is Age 65 Plus at Start"
+					then ("Is Enrolled in Institutional SNP"
+						or "Is Living Long-Term in Institution")
+					else false))
+		else false
+	end
+
+define "Any Patient Coverage At Any Time":
+	[Coverage] Cov
+		where Cov.status.value in { 'active', 'cancelled' }
+	// We assume [Coverage] in Patient context implicitly has this filter:
+	// Cov.beneficiary.identifier = Patient.identifier
+	// ... and that the paitient is not connected as a policy holder or payor.
+
+define "Product Line Organizations":
+	case "Product Line"
+		when 'commercial' then
+			[Organization] Org
+				where Org.name.value not in { 'Medicaid', 'Medicare' }
```

Comment:
Are we certain that this is how the Organization name will be represented? Should you also check aliases? Isn't it better/easier to check if the Coverage type code is SUBSUPP?

Reply:
The FHIR spec for Coverage and related resources was very unclear on how one would distinguish Medicaid or Medicare or Commercial and so I used this name check as a rough best guess.  I completely agree this test needs overhaul, but I consider resolution to be a similar problem to associating terminology with concrete identifiers, to be done after this pull request is merged.

Reply:
Agreed, this is a terminology question, but I would at the very least use the `type` element of the `Coverage` resource and set up a dummy value set for it.